django/db/models/base.py

```diff
@@ -1861,6 +1861,33 @@ def _check_constraints(cls, databases):
                 )
         return errors
 
+    @classmethod
+    def _warn_unique_constraints(cls, databases):
+        errors = []
+        for db in databases:
+            if not router.allow_migrate_model(db, cls):
+                continue
+            connection = connections[db]
+            if (
+                connection.features.supports_partial_indexes or
+                'supports_partial_indexes' in cls._meta.required_db_features
+            ):
+                continue
+            if any(isinstance(constraint, UniqueConstraint) for constraint in cls._meta.constraints):
+                errors.append(
+                    checks.Warning(
+                        "%s does not support partial indexing, so the unique constraint "
+                        "won't be created." % connection.display_name,
+                        hint=(
+                            "The constraint won't be created. Silence this "
+                            "warning if you don't care about it."
```

Comment:
The wording is a bit inconsistent with the one for check constraints, here there is some duplicate info between the warning and the hint, and it talks about "The constraint" without naming it. I would suggest:

```
                    checks.Warning(
                        "%s does not support unique constraints with conditions." % connection.display_name,
                        hint=(
                            "A constraint won't be created. Silence this "
                            "warning if you don't care about it."
```